Sources/Prometheus/MetricTypes/Histogram.swift

```diff
@@ -123,20 +123,33 @@ public class Histogram<NumType: DoubleRepresentable, Labels: HistogramLabels>: M
     /// - Parameters:
     ///     - value: Value to observe
     ///     - labels: Labels to attach to the observed value
-    public func observe(_ value: NumType, _ labels: Labels? = nil) {
+    ///     - done: Completion handler
+    public func observe(_ value: NumType, _ labels: Labels? = nil, _ done: @escaping () -> Void = { }) {
         prometheusQueue.async(flags: .barrier) {
             if let labels = labels, type(of: labels) != type(of: EmptySummaryLabels()) {
                 let his = self.prometheus.getOrCreateHistogram(with: labels, for: self)
-                his.observe(value)
+                his.observe(value) {
```

Comment:
Could you extract the stuff you have in the completion handler here into a function? So you don't need the code twice?
Something like:
```swift
public func observe(_ value: NumType, _ labels: Labels? = nil, _ done: @escaping () -> Void = { }) {
    prometheusQueue.async(flags: .barrier) {
        var comp: () -> Void = { 
            self.total.inc(value)
                    
            for (i, bound) in self.upperBounds.enumerated() {
                if bound >= value.doubleValue {
                    self.buckets[i].inc()
                    break
                }
            }
            done()
        }
        if let labels = labels, type(of: labels) != type(of: EmptySummaryLabels()) {
            let his = self.prometheus.getOrCreateHistogram(with: labels, for: self)
            return his.observe(value, comp)
        }
        return comp()
    }
}
```

Reply:
Much cleaner, thanks!